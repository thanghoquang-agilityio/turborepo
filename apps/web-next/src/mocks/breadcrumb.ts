export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
