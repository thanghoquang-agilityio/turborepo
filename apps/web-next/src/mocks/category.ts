export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
