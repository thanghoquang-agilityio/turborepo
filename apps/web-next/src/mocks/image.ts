import { SRC_IMAGE_NOT_AVAILABLE } from '@/constants'

export const MOCK_IMAGE = {
  data: {
    id: '24',
    attributes: {
      name: 'default',
      url: SRC_IMAGE_NOT_AVAILABLE,
    },
  },
}

export const MOCK_LIST_IMAGE_RESPONSE = [
  {
    name: 'default1',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default2',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default3',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
  {
    name: 'default4',
    url: SRC_IMAGE_NOT_AVAILABLE,
  },
]
