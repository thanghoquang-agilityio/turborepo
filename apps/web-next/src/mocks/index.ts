export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
