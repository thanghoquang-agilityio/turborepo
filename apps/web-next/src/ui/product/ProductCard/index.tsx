import { memo, useCallback } from 'react'

import Link from 'next/link'
import { useRouter } from 'next/navigation'

import { Card } from '@nextui-org/react'

// Components
import { Button, ButtonIcon, CustomImage, Text } from '@/components'
// Constants
import { API_IMAGE_URL, COMMENTS_DEFAULT, ROUTER } from '@/constants'
import { HeartIcon, StarIcon } from '@/icons'
// Styles
import '@/styles/card.css'
// Utils
import { formatAmountCurrency, getStars } from '@/utils'

interface ProductCardProps {
  id: string
  image?: string
  name: string
  variants?: number
  price: number
  star: number
  comments: number
  additionalClass?: string
}

const ProductCard = memo(
  ({
    id,
    image,
    name,
    variants,
    star,
    comments,
    price,
    additionalClass = '',
  }: ProductCardProps) => {
    const altImage = name.split('')[0]
    const formatPrice = formatAmountCurrency(price.toString())

    const router = useRouter()
    const handleAddToCart = useCallback(() => {
      router.push(`${ROUTER.PRODUCTS}/${id}`)
    }, [id, router])

    return (
      <Card
        className={`card relative flex max-w-[305px] flex-col items-center overflow-hidden rounded-large border-medium border-background-300 ${additionalClass}`}
      >
        <ButtonIcon
          variant='ghost'
          size='sm'
          shape='circle'
          additionalClass='absolute top-3 right-2 min-w-[30px]'
        >
          <HeartIcon />
        </ButtonIcon>
        <Link href={`${ROUTER.PRODUCTS}/${id}`}>
          <CustomImage
            src={`${API_IMAGE_URL}${image}`}
            alt={altImage}
            className='card-image w-full md:h-[303px] md:w-[305px]'
            width={305}
            height={303}
          />
        </Link>

        <div className='flex w-full flex-col gap-[5px] overflow-hidden p-5'>
          <div className='card-content flex flex-col justify-between md:flex-row'>
            <Text
              variant='primary'
              size='lg'
              className='max-w-[150px] text-primary-400'
            >
              {name}
            </Text>
            <Text variant='title' size='lg' className='max-w-[100px]'>
              {formatPrice}
            </Text>
          </div>
          <Text
            variant='subTitle'
            size='sm'
            className='max-w-[263px] text-secondary-500'
          >
            {variants} types of shoos available
          </Text>
          <div className='mt-2 flex gap-2'>
            <div className='gap-0.25 flex h-[17px] text-warning-200'>
              {getStars(star).map((value, index) => (
                <StarIcon key={`star-product-${id}-${value}-${index}`} />
              ))}
            </div>
            <Text variant='subTitle' size='sm' className='text-secondary-500'>
              ({comments || COMMENTS_DEFAULT})
            </Text>
          </div>
        </div>

        <div className='card-content mb-6 mt-1 flex w-full flex-col items-end gap-3 px-5 md:flex-row md:justify-between'>
          <Button
            variant='solid'
            size='sm'
            additionalClass='md:max-w-[128px] w-full'
            onClick={handleAddToCart}
            data-testid='btn-add-to-cart'
            aria-label='Add To Cart'
          >
            Add To Cart
          </Button>
          <Button
            variant='bordered'
            size='sm'
            additionalClass='md:max-w-[128px] w-full cursor-not-allowed'
            aria-label='Add Shortlist'
          >
            Add Shortlist
          </Button>
        </div>
      </Card>
    )
  }
)

ProductCard.displayName = 'ProductCard'
export default ProductCard
