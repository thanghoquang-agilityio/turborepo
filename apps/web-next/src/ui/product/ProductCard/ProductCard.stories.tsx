import type { Meta, StoryObj } from '@storybook/react'

import { MOCK_PRODUCT } from '@/mocks'
import ProductCard from '@/ui/product/ProductCard'

const meta = {
  title: 'Components/Card',
  tags: ['autodocs'],
  component: ProductCard,
  parameters: {
    layout: 'centered',
    nextjs: {
      appDirectory: true,
    },
  },
} satisfies Meta<typeof ProductCard>

export default meta
type Story = StoryObj<typeof ProductCard>

export const Default: Story = {
  args: { ...MOCK_PRODUCT, image: '' },
}
