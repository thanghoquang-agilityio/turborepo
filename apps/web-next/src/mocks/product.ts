import { ProductVariantModel } from '@/types'

import { MOCK_IMAGE } from './image'

// New flattened product variants for testing
export const MOCK_PRODUCT_VARIANTS: ProductVariantModel[] = [
  {
    id: 1,
    color: 'slate blue',
    size: 'small,medium,large,extra large,XXL',
    stock: '20,20,20,20,20',
    tax: 5,
  },
  {
    id: 2,
    color: 'blue',
    size: 'small,medium,large,extra large,XXL',
    stock: '20,20,20,20,20',
    tax: 5,
  },
]

export const MOCK_PRODUCT = {
  id: 1,
  documentId: 'mock-product-1',
  image: MOCK_IMAGE,
  name: 'TDX Sinkers',
  variants: 5,
  price: 675.0,
  star: 5,
  comments: 121,
  description: '',
  discount: 0,
  likes: 11,
  brand: 'MOCK BRAND',
  productVariants: MOCK_PRODUCT_VARIANTS,
}

export const MOCK_PRODUCTS_RESPONSE = [
  {
    id: 1,
    documentId: 'mock-product-1',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 2,
    documentId: 'mock-product-2',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 3,
    documentId: 'mock-product-3',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: [],
  },
  {
    id: 4,
    documentId: 'mock-product-4',
    name: 'TDX Sinkers',
    variants: 5,
    price: 675.0,
    star: 5,
    comments: 121,
    brand: 'MOCK BRAND',
    description: '',
    discount: 0,
    likes: 11,
    productVariants: MOCK_PRODUCT_VARIANTS,
    image: MOCK_IMAGE,
  },
]
