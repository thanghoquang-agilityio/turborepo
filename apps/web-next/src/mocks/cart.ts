import { CartItemResponse } from '@/types'
import { MOCK_PRODUCT_VARIANTS } from './product'

export const MOCK_CART_ITEM = {
  id: '1',
  variantId: '1',
  documentId: '1',
  name: 'Embrace Sideboard',
  color: 'wheat',
  size: 'small',
  quantityDefault: 2,
  stock: 4,
  price: 53,
  star: 1,
  comments: 1,
  tax: 1,
  discount: 1,
}

// New flattened cart response structure
export const MOCK_CART_RESPONSE: CartItemResponse[] = [
  {
    id: 1,
    documentId: 'cart-item-1',
    size: 'small',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[0],
    userId: 1,
  },
  {
    id: 2,
    documentId: 'cart-item-2',
    size: 'large',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[1],
    userId: 1,
  },
]

export const MOCK_ORDER_SUMMARY = {
  subTotal: 396,
  itemTotal: 3,
}
