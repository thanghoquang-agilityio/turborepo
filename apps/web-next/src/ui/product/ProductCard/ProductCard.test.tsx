import { fireEvent, render } from '@testing-library/react'

import { MOCK_CART_ITEM } from '@/mocks'

import ProductCard from '.'

const mockPush = jest.fn()

jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockImplementation(() => ({
    push: mockPush,
  })),
}))

describe('ProductCard', () => {
  const product = { ...MOCK_CART_ITEM, image: '' }
  const renderComponent = () => render(<ProductCard {...product} />)
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })

  it('should navigate to the product detail page when "Add To Cart" is clicked', () => {
    const { getByTestId } = renderComponent()
    fireEvent.click(getByTestId('btn-add-to-cart'))
    expect(mockPush).toHaveBeenCalledWith('/products/1')
  })
})
