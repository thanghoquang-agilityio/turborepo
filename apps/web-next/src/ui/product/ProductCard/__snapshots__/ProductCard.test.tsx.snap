// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should match snapshot 1`] = `
<div>
  <div
    class="h-auto text-foreground box-border bg-content1 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium transition-transform-background motion-reduce:transition-none card relative flex max-w-[305px] flex-col items-center overflow-hidden rounded-large border-medium border-background-300"
    tabindex="-1"
  >
    <button
      aria-label="Button Icon"
      class="z-0 group inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-background-100 border-none text-primary-500 rounded-[100%] sm:min-w-[46px] sm:min-h-[46px] sm:p-[11px] h-[30px] p-[7px] w-[30px] absolute top-3 right-2 min-w-[30px]"
      type="button"
    >
      <svg
        class="w-full h-full"
        fill="none"
        viewBox="0 0 26 26"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M22.1763 5.51296C21.652 4.98834 21.0294 4.57217 20.3441 4.28823C19.6588 4.0043 18.9244 3.85815 18.1826 3.85815C17.4408 3.85815 16.7064 4.0043 16.0211 4.28823C15.3358 4.57217 14.7132 4.98834 14.1889 5.51296L13.1006 6.60123L12.0123 5.51296C10.9531 4.45376 9.51654 3.8587 8.0186 3.8587C6.52066 3.8587 5.08407 4.45376 4.02486 5.51296C2.96566 6.57217 2.37061 8.00875 2.37061 9.50669C2.37061 11.0046 2.96566 12.4412 4.02486 13.5004L5.11313 14.5887L13.1006 22.5762L21.0881 14.5887L22.1763 13.5004C22.701 12.9761 23.1171 12.3535 23.4011 11.6682C23.685 10.9829 23.8311 10.2484 23.8311 9.50669C23.8311 8.76494 23.685 8.03045 23.4011 7.3452C23.1171 6.65994 22.701 6.03734 22.1763 5.51296V5.51296Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2.156"
        />
      </svg>
    </button>
    <a
      href="/products/1"
    >
      <img
        alt="E"
        class="card-image w-full md:h-[303px] md:w-[305px]"
        data-nimg="1"
        decoding="async"
        height="303"
        loading="lazy"
        src="http://localhost/_next/image?url=https%3A%2F%2Fe-commerce-api-2t2h.onrender.com&w=640&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fe-commerce-api-2t2h.onrender.com&w=384&q=75 1x, /_next/image?url=https%3A%2F%2Fe-commerce-api-2t2h.onrender.com&w=640&q=75 2x"
        style="color: transparent; object-fit: cover;"
        width="305"
      />
    </a>
    <div
      class="flex w-full flex-col gap-[5px] overflow-hidden p-5"
    >
      <div
        class="card-content flex flex-col justify-between md:flex-row"
      >
        <p
          class="text-primary-300 font-semibold text-lg no-underline text-ellipsis whitespace-nowrap overflow-hidden max-w-[150px] text-primary-400"
        >
          Embrace Sideboard
        </p>
        <p
          class="text-primary-200 dark:text-primary-300 font-semibold text-lg no-underline text-ellipsis whitespace-nowrap overflow-hidden max-w-[100px]"
        >
          $53.00
        </p>
      </div>
      <p
        class="text-foreground-300 font-normal text-sm no-underline text-ellipsis whitespace-nowrap overflow-hidden max-w-[263px] text-secondary-500"
      >
         types of shoos available
      </p>
      <div
        class="mt-2 flex gap-2"
      >
        <div
          class="gap-0.25 flex h-[17px] text-warning-200"
        >
          <svg
            class="w-full h-full"
            fill="none"
            viewBox="0 0 16 15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.94562 0.371195C8.09111 0.0214042 8.58663 0.0214039 8.73211 0.371195L10.492 4.60253C10.5534 4.74999 10.692 4.85074 10.8512 4.86351L15.4193 5.22973C15.7969 5.26 15.9501 5.73127 15.6623 5.97772L12.182 8.95905C12.0607 9.06295 12.0077 9.22598 12.0447 9.38133L13.1081 13.839C13.196 14.2075 12.7951 14.4988 12.4718 14.3013L8.56087 11.9125C8.42457 11.8293 8.25316 11.8293 8.11686 11.9125L4.20595 14.3013C3.88265 14.4988 3.48177 14.2075 3.56967 13.839L4.63298 9.38133C4.67004 9.22598 4.61707 9.06295 4.49578 8.95905L1.01539 5.97773C0.727672 5.73127 0.880795 5.26 1.25842 5.22973L5.8265 4.86351C5.9857 4.85074 6.12438 4.74999 6.18571 4.60253L7.94562 0.371195Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="w-full h-full"
            fill="none"
            viewBox="0 0 16 15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.94562 0.371195C8.09111 0.0214042 8.58663 0.0214039 8.73211 0.371195L10.492 4.60253C10.5534 4.74999 10.692 4.85074 10.8512 4.86351L15.4193 5.22973C15.7969 5.26 15.9501 5.73127 15.6623 5.97772L12.182 8.95905C12.0607 9.06295 12.0077 9.22598 12.0447 9.38133L13.1081 13.839C13.196 14.2075 12.7951 14.4988 12.4718 14.3013L8.56087 11.9125C8.42457 11.8293 8.25316 11.8293 8.11686 11.9125L4.20595 14.3013C3.88265 14.4988 3.48177 14.2075 3.56967 13.839L4.63298 9.38133C4.67004 9.22598 4.61707 9.06295 4.49578 8.95905L1.01539 5.97773C0.727672 5.73127 0.880795 5.26 1.25842 5.22973L5.8265 4.86351C5.9857 4.85074 6.12438 4.74999 6.18571 4.60253L7.94562 0.371195Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="w-full h-full"
            fill="none"
            viewBox="0 0 16 15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.94562 0.371195C8.09111 0.0214042 8.58663 0.0214039 8.73211 0.371195L10.492 4.60253C10.5534 4.74999 10.692 4.85074 10.8512 4.86351L15.4193 5.22973C15.7969 5.26 15.9501 5.73127 15.6623 5.97772L12.182 8.95905C12.0607 9.06295 12.0077 9.22598 12.0447 9.38133L13.1081 13.839C13.196 14.2075 12.7951 14.4988 12.4718 14.3013L8.56087 11.9125C8.42457 11.8293 8.25316 11.8293 8.11686 11.9125L4.20595 14.3013C3.88265 14.4988 3.48177 14.2075 3.56967 13.839L4.63298 9.38133C4.67004 9.22598 4.61707 9.06295 4.49578 8.95905L1.01539 5.97773C0.727672 5.73127 0.880795 5.26 1.25842 5.22973L5.8265 4.86351C5.9857 4.85074 6.12438 4.74999 6.18571 4.60253L7.94562 0.371195Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="w-full h-full"
            fill="none"
            viewBox="0 0 16 15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.94562 0.371195C8.09111 0.0214042 8.58663 0.0214039 8.73211 0.371195L10.492 4.60253C10.5534 4.74999 10.692 4.85074 10.8512 4.86351L15.4193 5.22973C15.7969 5.26 15.9501 5.73127 15.6623 5.97772L12.182 8.95905C12.0607 9.06295 12.0077 9.22598 12.0447 9.38133L13.1081 13.839C13.196 14.2075 12.7951 14.4988 12.4718 14.3013L8.56087 11.9125C8.42457 11.8293 8.25316 11.8293 8.11686 11.9125L4.20595 14.3013C3.88265 14.4988 3.48177 14.2075 3.56967 13.839L4.63298 9.38133C4.67004 9.22598 4.61707 9.06295 4.49578 8.95905L1.01539 5.97773C0.727672 5.73127 0.880795 5.26 1.25842 5.22973L5.8265 4.86351C5.9857 4.85074 6.12438 4.74999 6.18571 4.60253L7.94562 0.371195Z"
              fill="currentColor"
            />
          </svg>
          <svg
            class="w-full h-full"
            fill="none"
            viewBox="0 0 16 15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.94562 0.371195C8.09111 0.0214042 8.58663 0.0214039 8.73211 0.371195L10.492 4.60253C10.5534 4.74999 10.692 4.85074 10.8512 4.86351L15.4193 5.22973C15.7969 5.26 15.9501 5.73127 15.6623 5.97772L12.182 8.95905C12.0607 9.06295 12.0077 9.22598 12.0447 9.38133L13.1081 13.839C13.196 14.2075 12.7951 14.4988 12.4718 14.3013L8.56087 11.9125C8.42457 11.8293 8.25316 11.8293 8.11686 11.9125L4.20595 14.3013C3.88265 14.4988 3.48177 14.2075 3.56967 13.839L4.63298 9.38133C4.67004 9.22598 4.61707 9.06295 4.49578 8.95905L1.01539 5.97773C0.727672 5.73127 0.880795 5.26 1.25842 5.22973L5.8265 4.86351C5.9857 4.85074 6.12438 4.74999 6.18571 4.60253L7.94562 0.371195Z"
              fill="currentColor"
            />
          </svg>
        </div>
        <p
          class="text-foreground-300 font-normal text-sm no-underline text-ellipsis whitespace-nowrap overflow-hidden text-secondary-500"
        >
          (
          1
          )
        </p>
      </div>
    </div>
    <div
      class="card-content mb-6 mt-1 flex w-full flex-col items-end gap-3 px-5 md:flex-row md:justify-between"
    >
      <button
        aria-label="Add To Cart"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-primary-300 text-background-100 py-2 min-w-32 h-9 text-sm rounded-[30px] md:max-w-[128px] w-full"
        data-testid="btn-add-to-cart"
        type="button"
      >
        Add To Cart
      </button>
      <button
        aria-label="Add Shortlist"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-background-100 text-primary-100 border-small border-foreground-300 py-2 min-w-32 h-9 text-sm rounded-[30px] md:max-w-[128px] w-full cursor-not-allowed"
        type="button"
      >
        Add Shortlist
      </button>
    </div>
  </div>
</div>
`;
